pkg/backtest/exchange.go

```diff
@@ -362,21 +362,36 @@ func (e *Exchange) SubscribeMarketData(startTime, endTime time.Time, extraInterv
 	return klineC, nil
 }
 
-func (e *Exchange) ConsumeKLine(k types.KLine) {
-	if k.Interval == types.Interval1m {
-		e.currentTime = k.EndTime.Time()
-
-		matching, ok := e.matchingBook(k.Symbol)
-		if !ok {
-			log.Errorf("matching book of %s is not initialized", k.Symbol)
-			return
+func (e *Exchange) ConsumeKLine(k types.KLine, handlers []func(types.KLine, *ExchangeDataSource), src *ExchangeDataSource) {
```

Comment:
Is there any way to make this cleaner? I don't want to move the kline handlers into this method, it makes it harder to be refactored.

Reply:
you could:
1. either expand the caching code in cmd/backtest.go
2. add api to query handlers from ExchangeDataSource during the runtime